middleware.go

```diff
@@ -139,7 +145,30 @@ func (tw traceware) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		oteltrace.WithAttributes(semconv.HTTPServerAttributesFromHTTPRequest(tw.serverName, routePattern, r)...),
 		oteltrace.WithSpanKind(oteltrace.SpanKindServer),
 	)
-	defer span.End()
+	defer func() {
+		if recovered := recover(); recovered != nil {
+			defer panic(recovered)
+
+			span.SetStatus(semconv.SpanStatusFromHTTPStatusCode(http.StatusInternalServerError))
+
+			opts := []oteltrace.EventOption{
+				oteltrace.WithAttributes(
+					semconv.ExceptionTypeKey.String(typeStr(recovered)),
+					semconv.ExceptionMessageKey.String(fmt.Sprint(recovered)),
+				),
+			}
+
+			if tw.recordStackTrace {
+				opts = append(opts, oteltrace.WithAttributes(
+					semconv.ExceptionStacktraceKey.String(recordStackTrace()),
+				))
+			}
```

Comment:
Have you check if this attribute recorded properly?
I've create a test to check this, but those attributes not found in the `span`.

You can see the test in here
https://github.com/knightazura/otelchi/blob/59d949aa1b25b6b3230f7ac09e352f11c2a50e46/middleware_test.go#L378-L437